core/services/llo/channeldefinitions/onchain_channel_definition_cache.go

```diff
@@ -192,11 +241,87 @@ func (c *channelDefinitionCache) Start(ctx context.Context) error {
 	})
 }
 
-////////////////////////////////////////////////////////////////////
-// Log Polling
-////////////////////////////////////////////////////////////////////
+// blockNumFromUint64 converts a uint64 block number to int64
+// This is safe as block numbers are well within int64 range
+func blockNumFromUint64(blockNum uint64) int64 {
+	//nolint:gosec // disable G115
+	return int64(blockNum)
+}
+
+// unpackOwnerLog unpacks and validates an owner log from logpoller
+// Returns the unpacked log and an error if unpacking or validation fails
+func (c *channelDefinitionCache) unpackOwnerLog(log logpoller.Log) (*channel_config_store.ChannelConfigStoreNewChannelDefinition, error) {
```

Comment:
Is it possible to use or transform `logpoller.Log` into the type that [ParseNewChannelDefinition](https://github.com/smartcontractkit/chainlink-evm/blob/da6e4a9838e94e56df48a3dd2daf8429a698742e/gethwrappers/llo-feeds/generated/channel_config_store/channel_config_store.go#L397-L404) uses so we can avoid all the logic below?